File: src/clase-05/practica/typescript/activity2_adapter_starter/adapter.ts
// Interfaz objetivo
export interface NewInterface {
  request(): string;
}

// Clase existente (incompatible)
export class OldSystem {
  specificRequest(): string {
    return "<data><msg>Respuesta del sistema antiguo</msg></data> (XML → datos brutos)";
  }
}

// Adapter que traduce a la interfaz objetivo (XML → JSON simulado)
export class Adapter implements NewInterface {
  constructor(private adaptee: OldSystem) {}

  request(): string {
    const raw = this.adaptee.specificRequest();
    return raw
      .replace(" (XML → datos brutos)", " (normalizado)")
      .replace("<data><msg>", "{'msg':'")
      .replace("</msg></data>", "'}");
  }
}

File: src/clase-05/practica/python/activity2_adapter_starter/adapter.py
from abc import ABC, abstractmethod
from old_system import OldSystem

class NewInterface(ABC):
    @abstractmethod
    def request(self) -> str:  # pragma: no cover
        pass

class Adapter(NewInterface):
    def __init__(self, adaptee: OldSystem):
        self._adaptee = adaptee

    def request(self) -> str:
        # TODO: traducir la interfaz del adaptee y normalizar el formato (XML → JSON simulado)
        raw = self._adaptee.specific_request()
        normalized = raw.replace(" (XML → datos brutos)", " (normalizado)")                        .replace("<data><msg>", "{'msg':'")                        .replace("</msg></data>", "'}")
        return normalized
